unity/Assets/Maroon/scenes/experiments/OpticsSimulations/Scripts/Manager/LightComponentManager.cs

```diff
@@ -0,0 +1,138 @@
+using System.Collections.Generic;
+using Maroon.scenes.experiments.OpticsSimulations.Scripts.TableObject.Handlers;
+using Maroon.scenes.experiments.OpticsSimulations.Scripts.TableObject.LightComponent;
+using Maroon.scenes.experiments.OpticsSimulations.Scripts.Util;
+using UnityEngine;
+using LaserPointer = Maroon.scenes.experiments.OpticsSimulations.Scripts.TableObject.LightComponent.LaserPointer;
+
+namespace Maroon.scenes.experiments.OpticsSimulations.Scripts.Manager
+{
+    public class LightComponentManager : MonoBehaviour
+    {
+        public static LightComponentManager Instance;
+
+        private List<LightComponent> _lightComponents;
+        [SerializeField] private GameObject tableLowLeftCorner;
+        
+        [Header("Prefabs: Light Sources")] 
+        [SerializeField] private LaserPointer laserPointer;
+        [SerializeField] private ParallelSource parallelSource;
+        [SerializeField] private PointSource pointSource;
+
+        public List<LightComponent> LightComponents => _lightComponents;
+        
+
+        private void Awake()
+        {
+            if (Instance == null)
+                Instance = this;
+            else
+            {
+                Debug.LogError("SHOULD NOT OCCUR - Destroyed LightSourceManager");
+                Destroy(gameObject);
+            }
+            _lightComponents = new List<LightComponent>();
+        }
+
+        public LightComponent AddLightComponent(LightComponent lc, Vector3 pos, Vector3? rot = null, List<float> wavelengths = null)
+        {
+            Vector3 rotation = rot ?? new Vector3(0, 0, 0);
+            if (wavelengths == null)
+                wavelengths = new List<float> {720f};
+            
+            var lsClone = Instantiate(lc, tableLowLeftCorner.transform);
+            lsClone.transform.localPosition = pos;
+            lsClone.transform.right = rotation.normalized;
+            lsClone.Wavelengths = wavelengths;
+            _lightComponents.Add(lsClone);
+
+            return lsClone;
+        }
+
+        public void AddLC(int nr)
+        {
+            switch (nr)
+            {
+                case 0:
+                    return;
+                case 1:
+                    AddLightComponent(laserPointer, Constants.BaseLcPosition, wavelengths: laserPointer.Wavelengths);
+                    break;
+                case 2:
+                    AddLightComponent(parallelSource, Constants.BaseLcPosition, wavelengths: parallelSource.Wavelengths);
+                    break;
+                case 3:
+                    AddLightComponent(pointSource, Constants.BaseLcPosition, wavelengths: pointSource.Wavelengths);
+                    break;
+            }
+        }
+        
+        public void WavelengthMenu(int nr)
+        {
+            List<float> wls = new List<float>();
+            switch (nr)
```

Comment:
I think we can define a read-only array for the wavelength settings and the integer nr as intex to select it from the array.